### DISCUSSION TITLE
Simulating a Photon

### DISCUSSION THREAD

**Discussion by Ghiraa** [mitsuba-renderer/mitsuba3#1060] [Q&A]
Hi, I would like to start by mentioning I am a complete beginner at Mitsuba. I am using it for my final-year project. What I am trying to do is to have multiple sensors along a cylinder and check if I can capture a photon on only one sensor. Currently, I created a cylinder, and placed along its edge sensors equally distributed with their height equal to the one of the cylinder. To simulate a photon, I am using a spotlight with a narrow but powerful beam which points to one part of the cylinder. Although light is captured by only one sensor(which is good), I feel like something is wrong. Why is that square that big? Also, what's going on with the green bit? Btw, I am using 'pixel_format': 'luminance'. I guess somehow that's some light that's not as intense as the yellow bit, but if I do the beam narrower I can't capture it no matter how much is the intensity.

 This is what each sensor captures:
![image](https://github.com/mitsuba-renderer/mitsuba3/assets/45585468/48ca5d4c-1652-40f8-af2a-167baa50d178)

And this is how they look like concatenated:
![image](https://github.com/mitsuba-renderer/mitsuba3/assets/45585468/ecedcf52-01e8-424c-a4e4-55379370a9e3)

Thank you!

**Comment by njroussel** [comment]
Hi @Ghiraa 

I'm not totally sure what is going on either. My guess would be that the `spot` is indeed too wide and currently occupies two pixels.
Also, make sure you're using a `box` reconstruction filter on your film.

I'm assuming you're using the `ptracer` integrator?
It shouldn't be too hard to replicate some of its logic to debug what is happening here - or you can directly add prints into the integrator if you're compiling the project yourself and using `scalar_rgb'. Particularly, this snippet: https://github.com/mitsuba-renderer/mitsuba3/blob/master/src/integrators/ptracer.cpp#L143-L159. In short, it picks a point on the emitter (in your case the `spot` point), and tries to connect it to the sensor. 

> Currently, I created a cylinder, and placed along its edge sensors equally distributed with their height equal to the one of the cylinder

Just to double-check, what do you mean by sensor's height? Its field of view?

**Reply by Ghiraa** [reply]
I was not using ptracer :). I solve the issue thank you!
